storm-core/test/jvm/org/apache/storm/scheduler/blacklist/TestBlacklistScheduler.java

```diff
@@ -0,0 +1,326 @@
+                Set<String> hosts = new HashSet<>();
+                hosts.add("host-0");
+                Assert.assertEquals("blacklist", hosts, cluster.getBlacklistedHosts());
+            } else if (count == 30) {
+                Set<String> hosts = new HashSet<>();
+                hosts.add("host-0");
+                Assert.assertEquals("blacklist", hosts, cluster.getBlacklistedHosts());
+            } else if (count == 31) {
+                Set<String> hosts = new HashSet<>();
+                hosts.add("host-0");
+                hosts.add("host-1");
+                Assert.assertEquals("blacklist", hosts, cluster.getBlacklistedHosts());
+            } else if (count == 32) {
+                Set<String> hosts = new HashSet<>();
+                hosts.add("host-0");
+                hosts.add("host-1");
+                Assert.assertEquals("blacklist", hosts, cluster.getBlacklistedHosts());
+            } else if (count == 60) {
+                Set<String> hosts = new HashSet<>();
+                hosts.add("host-0");
+                hosts.add("host-1");
+                Assert.assertEquals("blacklist", hosts, cluster.getBlacklistedHosts());
+            } else if (count == 61) {
+                Set<String> hosts = new HashSet<>();
+                hosts.add("host-0");
+                Assert.assertEquals("blacklist", hosts, cluster.getBlacklistedHosts());
+            } else if (count == 62) {
+                Set<String> hosts = new HashSet<>();
+                Assert.assertEquals("blacklist", hosts, cluster.getBlacklistedHosts());
+            }
+            count++;
+        }
+
+    }
+
+    @Test
+    public void removeLongTimeDisappearFromCache(){
+        INimbus iNimbus=new TestUtilsForBlacklistScheduler.INimbusTest();
+
+        Map<String, SupervisorDetails> supMap=TestUtilsForBlacklistScheduler.genSupervisors(3,4);
+
+        Config config = new Config();
+        config.putAll(Utils.readDefaultConfig());
+        config.put(Config.BLACKLIST_SCHEDULER_TOLERANCE_TIME,200);
+        config.put(Config.BLACKLIST_SCHEDULER_TOLERANCE_COUNT,2);
+        config.put(Config.BLACKLIST_SCHEDULER_RESUME_TIME,300);
+        Map<String, TopologyDetails> topoMap = new HashMap<String, TopologyDetails>();
+
+        TopologyDetails topo1 = TestUtilsForBlacklistScheduler.getTopology("topo-1", config, 5, 15, 1, 1, currentTime - 2,true);
+        topoMap.put(topo1.getId(), topo1);
+
+        Cluster cluster = new Cluster(iNimbus, supMap, new HashMap<String, SchedulerAssignmentImpl>(), config);
+        Topologies topologies = new Topologies(topoMap);
+        BlacklistScheduler bs=new BlacklistScheduler(new DefaultScheduler());
+        bs.prepare(config);
+        bs.schedule(topologies,cluster);
+        cluster = new Cluster(iNimbus, TestUtilsForBlacklistScheduler.removeSupervisorFromSupervisors(supMap,"sup-0"),TestUtilsForBlacklistScheduler.assignmentMapToImpl(cluster.getAssignments()), config);
+        for(int i=0;i<20;i++){
```

Comment:
Let's put a comment why we call schedule() exactly 20 times here. (I mean what exactly
the `20` means.)